File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        StringBuilder sb = new StringBuilder();
        String answer = "";
        
        String[] arr = new String[numbers.length];
        
        for (int i=0;i<numbers.length;i++) {
            arr[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(arr, (s1,s2) -> (s2+s1).compareTo(s1+s2));
        
        if (arr[0].equals("0")) return "0";
        
        for (String s : arr){
            sb.append(s);
        }
        
        
        return sb.toString();
    }
    
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <bits/stdc++.h>

using namespace std;

vector<string> v;

bool cmp(string a, string b){
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    string answer = "";
    
    for (int n : numbers) {
        v.push_back(to_string(n));
    }
    
    sort(v.begin(), v.end(), cmp);
    
    
    for (string s : v) answer += s;
    
    return answer[0] == '0' ? "0" : answer;
}